### PR TITLE
[dockerng] Migrate salt.mine to dockerng

### PR DESCRIPTION
**Pull request by ticosax**
Take advantage of the new wonderful dockerng module for mining feature.

This pull request consist into bringing back `host`argument from `docker.get_containers` to `dockerng.ps`. And, to trig a mine refresh on every salt function that might trig a state change.

/cc @terminalmage

### CODE REVIEW

**Comment by rallytime** [comment]
@ticosax Would you mind grabbing the pylint error really quickly? 

https://jenkins.saltstack.com/job/salt-pr-lint-n/5907/violations/file/salt/modules/mine.py/

**Comment by ticosax** [comment]
@rallytime thx for the notification. I fixed it.

**Comment by rallytime** [comment]
Awesome! Thank you!

**Comment by thatch45** [comment]
@terminalmage please let me know your thoughts on this one. I am all for it, but I know you has reservations earlier

**Comment by ticosax** [comment]
Is there is anything else I can do to help this PR to be merged ?

**Comment by terminalmage** [comment]
Sorry, I've had a lot to do any haven't had a chance to review this at all, and I think my review is what is holding this up. I will try to make some time to review it today, OK?

File: salt/modules/dockerng.py [hunk 1]
```diff
@@ -2207,6 +2222,9 @@ def ps_(**kwargs):
     all : False
         If ``True``, stopped containers will also be returned
 
+    host: False
+        If ``True``, local host's network topoly will be included
```
Comment (terminalmage) [inline salt/modules/dockerng.py]: Spelling error here I think, should be "topology".

**Comment by ticosax** [comment]
@terminalmage thank you It is perfectly fine today.
It is the last PR of the series I made those last few weeks.
Once it is merged I will give it a try with wider audience and a real use case with my testing environment.

File: salt/modules/dockerng.py (hunk 1 above)
Comment (ticosax) [inline salt/modules/dockerng.py]: Good catch

**Comment by terminalmage** [comment]
@ticosax I don't see anything major that jumps out at me here, aside from the fact that it seems the mine integration has been moved from the old docker module to the new module, which would remove functionality from users' existing deployments. As long as @thatch45 has no problem with the mine integration shifting to dockerng, I'm OK with it, but I think we should probably also add a notice in the relevant docstring(s) in the mine module (via a ``versionchanged`` RST statement) so that this is properly documented for users.

**Comment by ticosax** [comment]
Thanks, I see what you mean,
I'll try to not break mine integration with existing `modules/dockerio.py`.
working on it

**Comment by ticosax** [comment]
I pushed a proposal.
`dockerio.py` is using `dockerng.ps` to send mine data.

**Comment by jtand** [comment]
Go Go Jenkins!

**Comment by jtand** [comment]
Go Go Jenkins!

**Comment by ticosax** [comment]
rebased against develop. Go Go Jenkins ?

**Comment by jfindlay** [comment]
Go Go Jenkins!

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by ticosax** [comment]
has been rebased.

**Comment by jfindlay** [comment]
Nice, thanks @ticosax.

**Comment by ticosax** [comment]
thank you ! :sweat_smile:

**Pull request merged**